src/libtiled/tilelayer.h

```diff
@@ -191,6 +244,8 @@ class TILEDSHARED_EXPORT TileLayer : public Layer
     bool contains(int x, int y) const;
     bool contains(const QPoint &point) const;
 
+    QPair<int, int> block(int x, int y) const;
```

Comment:
Yes, though that would require doing the division with doubles and converting back to integer. Alternatively you could special-case negative values:
```c++
x < 0 ? (x + 1) / CHUNK_SIZE - 1 : x / CHUNK_SIZE
```
I'm not sure what's best performance-wise, but I would guess this one to be faster.